src/System.Management.Automation/resources/DiscoveryExceptions.resx

```diff
@@ -191,6 +193,9 @@ The #requires statement must be in one of the following formats:
   <data name="RequiresElevation" xml:space="preserve">
     <value>The script '{0}' cannot be run because it contains a "#requires" statement for running as Administrator. The current PowerShell session is not running as Administrator. Start PowerShell by using the Run as Administrator option, and then try running the script again.</value>
   </data>
+  <data name="RequiresOSTypeInvalid" xml:space="preserve">
+    <value>The script '{0}' cannot be run because your current OS type {1} does not satisfy the "#requires" statement for running on OS types: {2}.</value>
```

Comment:
```suggestion
    <value>The script '{0}' cannot be run because the current operating system type '{1}' does not satisfy the "#requires" statement for running on operating system types: '{2}'.</value>
```